### PR TITLE
INSTUI-2932 : Replace <Flex> with <span> in Button to improve performance

### PR DESCRIPTION
**Pull request by matyasf**
Replace the Flex component with simple span inside BaseButton for better performance. In the testing tool it resulted in ~40% performance improvement testing 100 times with a Button that has an icon and text:

BUTTON_ICON_IUIx100	832.9ms  // old, 8.4.0 code
BUTTON_ICON_IUIx100	504.4ms  // new code in this PR

It should improve performance with other types of button too, albeit to a lesser degree

### CODE REVIEW

**Review by Brailor — APPROVED** [review]

File: packages/ui-buttons/src/BaseButton/index.tsx [hunk 1]
```diff
@@ -356,38 +355,21 @@ class BaseButton extends Component<Props> {
     )
 
     const flexChildren = hasOnlyIconVisible ? (
-      <Flex.Item>
+      <span css={styles.flexOnlyIcon}>
```
Comment (ke1k0) [inline packages/ui-buttons/src/BaseButton/index.tsx:358]: Please consider the renaming of the classes I sent you on slack.

File: packages/ui-buttons/src/BaseButton/styles.ts [hunk 2]
```diff
@@ -356,6 +356,42 @@ const generateStyle = (componentTheme, props, state) => {
 
       // @ts-expect-error ts-migrate(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
       ...sizeVariants[size].iconSVG
+    },
+
+    flex: {
```
Comment (ke1k0) [inline packages/ui-buttons/src/BaseButton/styles.ts:361]: Please add the missing css props from view (sent google sheet on slack)

File: packages/ui-buttons/src/BaseButton/index.tsx (hunk 1 above)
Comment (matyasf) [reply packages/ui-buttons/src/BaseButton/index.tsx]: done

File: packages/ui-buttons/src/BaseButton/styles.ts (hunk 2 above)
Comment (matyasf) [reply packages/ui-buttons/src/BaseButton/styles.ts]: done

**Review by ke1k0 — APPROVED** [review]

**Pull request merged**